app/api-client.js

```diff
@@ -134,10 +137,15 @@ module.exports = (function () {
     return api_request({method: 'POST', endpoint: 's3buckets', resource: bucket});
   };
 
+  api.grant_user_access_to_bucket = function (users3bucket) {
```

Comment:
This function signature is confusing, it should either be:
```javascript
api.grant_user_access_to_bucket = function (user, bucket) {
```
or
```javascript
api.create_users3bucket = function (users3bucket) {
```
Also, the implementation detail that the data source is an S3 bucket should not be revealed in these methods, because the frontend should not need to know about S3 or buckets.

Reply:
I will rename the *api* function to `api.create_users3bucket()` to make it more explicit and I may play with some ideas to make the API a bit nicer then 👍